internal/iam/repository_role_grant.go

```diff
@@ -536,3 +562,67 @@ func (m *MultiGrantTuple) TestStableSort() {
 	sort.Strings(gts)
 	m.Grants = strings.Join(gts, "^")
 }
+
+// grantsForUserGlobalResources returns the grants for the user for resources that can
+// only be globally scoped.
+func (r *Repository) grantsForUserGlobalResources(
+	ctx context.Context,
+	userId string,
+	res resource.Type,
+	opt ...Option,
+) (perms.GrantTuples, error) {
+	const op = "iam.(Repository).grantsForUserGlobalResources"
+	if userId == "" {
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing user id")
+	}
+
+	var args []any
+	switch userId {
+	case globals.AnonymousUserId:
+		args = append(args, sql.Named("user_ids", fmt.Sprintf("{ %s }", userId)))
+	default:
+		args = append(args, sql.Named("user_ids", fmt.Sprintf("{ u_anon, u_auth, %s }", userId)))
```

Comment:
We should probably use the consts for `u_auth` and `u_anon`, ie:

```suggestion
		args = append(args, sql.Named("user_ids", fmt.Sprintf("{ %s, %s, %s }", globals.AnonymousUserId, globals.AnyAuthenticatedUserId, userId)))
```